pkg/kubelet/pod_workers.go

```diff
@@ -1519,8 +1522,18 @@ func (p *podWorkers) completeWork(podUID types.UID, phaseTransition bool, syncEr
 		// Network is not ready; back off for short period of time and retry as network might be ready soon.
 		p.workQueue.Enqueue(podUID, wait.Jitter(backOffOnTransientErrorPeriod, workerBackOffPeriodJitterFactor))
 	default:
-		// Error occurred during the sync; back off and then retry.
-		p.workQueue.Enqueue(podUID, wait.Jitter(p.backOffPeriod, workerBackOffPeriodJitterFactor))
+		// Error occurred during the sync; back off and then retry. If the error includes a backoff expiration,
+		// wait until then instead of the default to avoid adding extra time to the backoff.
+		backoff := p.backOffPeriod
+		if backoffAt, isBackoffErr := kubecontainer.MinBackoffExpiration(syncErr); isBackoffErr {
+			backoff = backoffAt.Sub(p.clock.Now())
+		}
+		if backoff < 0 {
+			backoff = 0
+		} else if backoff > maxEnqueueDuration {
+			backoff = maxEnqueueDuration
```

Comment:
maybe we should just use the default sync interval (`p.resyncInterval`) as the max? That is the interval that we want to be checking the pod in steady-state, so I think it makes sense as a cap.

Reply:
The resync interval is usually 1 minute, so it's effectively the same as what you have.

Reply:
Removed `maxEnqueueDuration` and used `p.resyncInterval` instead.